ui/pages/swaps/awaiting-signatures/awaiting-signatures.js

```diff
@@ -0,0 +1,98 @@
+} from '../../../ducks/swaps/swaps';
+import PulseLoader from '../../../components/ui/pulse-loader';
+import SwapsFooter from '../swaps-footer';
+import SwapStepIcon from './swap-step-icon';
+
+export default function AwaitingSignatures() {
+  const t = useContext(I18nContext);
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const fetchParams = useSelector(getFetchParams);
+  const { destinationTokenInfo, sourceTokenInfo } = fetchParams?.metaData || {};
+  const approveTxParams = useSelector(getApproveTxParams);
+  const needsTwoConfirmations = Boolean(approveTxParams);
+
+  const awaitingSignaturesEvent = useNewMetricEvent({
+    event: 'Awaiting Signature(s) on a HW wallet',
+    sensitiveProperties: {
+      needs_two_confirmations: needsTwoConfirmations,
+      token_from: sourceTokenInfo?.symbol,
+      token_from_amount: fetchParams?.value,
+      token_to: destinationTokenInfo?.symbol,
+      request_type: fetchParams?.balanceError ? 'Quote' : 'Order',
+      slippage: fetchParams?.slippage,
+      custom_slippage: fetchParams?.slippage === 2,
+    },
+    category: 'swaps',
+  });
+
+  useEffect(() => {
+    awaitingSignaturesEvent();
+  }, [awaitingSignaturesEvent]);
+
+  const headerText = needsTwoConfirmations
+    ? t('swapTwoTransactions')
+    : t('swapConfirmWithHwWallet');
+
+  return (
+    <div className="awaiting-signatures">
+      <div className="awaiting-signatures__content">
+        <div className="awaiting-signatures__status-image">
+          <PulseLoader />
+        </div>
+        <div className="awaiting-signatures__header">{headerText}</div>
+        {needsTwoConfirmations && (
+          <>
+            <p className="awaiting-signatures__subheader">
+              {t('swapToConfirmWithHwWallet')}
+            </p>
+            <ul className="awaiting-signatures__steps">
+              <li>
+                <SwapStepIcon stepNumber={1} />
+                {t('swapAllowSwappingOf', [
+                  <span
+                    className="awaiting-signatures__symbol"
+                    key="allowToken"
+                  >
+                    {destinationTokenInfo.symbol}
+                  </span>,
+                ])}
+              </li>
+              <li>
+                <SwapStepIcon stepNumber={2} />
+                {t('swapFromTo', [
+                  <span className="awaiting-signatures__symbol" key="tokenFrom">
+                    {sourceTokenInfo.symbol}
+                  </span>,
+                  <span className="awaiting-signatures__symbol" key="tokenTo">
+                    {destinationTokenInfo.symbol}
+                  </span>,
+                ])}
+              </li>
+            </ul>
+            <div className="awaiting-signatures__main-description">
+              {t('swapGasFeesSplit')}
+            </div>
+          </>
+        )}
+      </div>
+      <SwapsFooter
+        onSubmit={async () => {
+          await dispatch(navigateBackToBuildQuote(history));
```

Comment:
Right now it goes to the `Awaiting Swap` page instead of `Build Quote` page, so I need to figure out why it doesn't clean up the current transaction when calling the `navigateBackToBuildQuote` function. It has maybe something to do with an `await` call for a HW wallet confirmation.

Reply:
Resolved.